core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -131,23 +143,12 @@ export class LoggedInUser extends BaseUser {
       this.page.waitForNavigation(),
       await this.clickOn(button),
     ]);
-    if (this.page.url() !== expectedDestinationPageUrl) {
-      throw new Error(
-        'The ' +
-          buttonName +
-          ' does not open the ' +
-          expectedDestinationPageName +
-          ' page!'
-      );
-    } else {
-      showMessage(
-        'The ' +
-          buttonName +
-          ' opens the ' +
-          expectedDestinationPageName +
-          ' page.'
-      );
-    }
+
+    expect(this.page.url())
+      .withContext(
+        `${buttonName} should open the ${expectedDestinationPageName} page`
```

Comment:
@seanlip, I'm not sure if we should show a message every time a test passes. Based on the result (the screenshot), the word "Passed" in green already shows a test passes or not. I think that it might not be that necessary to print a message when a test passes. If so, then we need to remove all the parts where it prints a message when a test passes, for the consistency.

Reply:
It's a good question. I'm happy either way as long as we're consistent. Maybe ask the rest of the acceptance test team and see what they think (since you'll all have a better opinion on whether or not it's useful)?

The pros: it's probably useful for quick debugging since you can see what statements triggered
The cons: a bit more bookkeeping for the dev to do, and might be clutter if it's hard to distinguish success statements from failure statements (though this is easy enough to do if we always prepend something like "ERROR:" to the latter).

Personally I lean slightly towards keeping the "passing" messages, but maybe start a discussion on chat and see what folks think?

Reply:
> The pros: it's probably useful for quick debugging since you can see what statements triggered

In practice, we generally just have these messages at the end of tests. As Christie pointed out, we already have the PASSED message with the spec name. So long as we make sure the spec names are descriptive enough, it serves the same purpose.

> The cons: a bit more bookkeeping for the dev to do, and might be clutter if it's hard to distinguish success statements from failure statements (though this is easy enough to do if we always prepend something like "ERROR:" to the latter)

I agree about the clutter. I don't think there will be much confusion on where failures occurred though.

![image](https://github.com/oppia/oppia/assets/9313126/95f98a3a-9861-4d18-872b-987e92ff5df7)

I am okay with discussing this more though.